File: 17_03_2024_Insert_Interval.py
class Solution:
    def insert(self, intervals, newInterval):
        ans = []
        x = newInterval[0]
        y = newInterval[1]
        i = 0
        n = len(intervals)
        while i < n:
            if intervals[i][1] < x:
                ans.append(intervals[i])
            elif intervals[i][0] > y:
                break
            else:
                x = min(intervals[i][0], x)
                y = max(intervals[i][1], y)
            i += 1
        ans.append([x, y])
        while i < n:
            ans.append(intervals[i])
            i += 1
        return ans

File: 17_03_2024_Insert_Interval.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> insert(List<List<Integer>> intervals, List<Integer> newInterval) {
        List<List<Integer>> ans = new ArrayList<>();
        int x = newInterval.get(0);
        int y = newInterval.get(1);
        int i = 0;
        int n = intervals.size();
        while (i < n) {
            if (intervals.get(i).get(1) < x) {
                ans.add(intervals.get(i));
            } else if (intervals.get(i).get(0) > y) {
                break;
            } else {
                x = Math.min(intervals.get(i).get(0), x);
                y = Math.max(intervals.get(i).get(1), y);
            }
            i++;
        }
        ans.add(List.of(x, y));
        while (i < n) {
            ans.add(intervals.get(i));
            i++;
        }
        return ans;
    }
}

